### PR TITLE
Add pricefeed content

### PR DESCRIPTION
**Pull request by satawatnack**
Details

- Add pricefeed docs (intro, get start and use-case)

### CODE REVIEW

File: docs/integration-tools/pricefeed/introduction.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+After BandChain processes the request, it will send an acknowledgement message along with `request_id` back. And, when the result of the request is finalized, BandChain will send a new IBC packet that contains the final data back. Relayers will listen and pick up those packets and relay them to the smart contract.
+
+After this stage, the cosmos-sdk app can safely use the data obtained from BandChain in its application at every `n` block interval. If the cosmos-sdk app requires additional data, it can submit an update symbols request proposal at any time.
+
+## pricefeed
+
+The pricefeed module obtains price data from BandChain through IBC and stores the most recent prices on your Cosmos SDK applications.
+
+An example of the usage of this module is provided on the [ConsumerChain](https://).
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/introduction.md]: please put url.

**Comment by colmazia** [comment]
change target branch to cw-band

File: docs/integration-tools/pricefeed/introduction.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+# Introduction
+
+The [cosmos-sdk](https://docs.cosmos.network/main/intro/overview) is presently the most widely utilized framework for developing blockchain applications. A crucial necessity for such apps is an oracle. As decentralized applications (dApps) rely heavily on real-world data, ensuring the accuracy of this data holds significant importance.
+
+It would be advantageous for your cosmos SDK app to possess an built-in oracle that obtains data from BandChain via IBC. By importing the pricefeed module implemented by Band Protocol.
+
+## Workflow
+![pricefeed](https://user-images.githubusercontent.com/32817745/226537402-8a76a57d-41e8-4299-ad8d-11521e31504d.png)
+
+### Proposal
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/introduction.md]: I think the proposal should be before the photo

File: docs/integration-tools/pricefeed/introduction.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+The Proposal submitted to update tasks for the pricefeed module consists of three components - the name of the symbol, the oracle script ID required to obtain the price, and the block interval for requesting the data every `n` block.
+
+Upon the proposal's approval, the pricefeed module will request price data from BandChain based on the SymbolRequest that was updated through the proposal. 
+
+
+### Request
+
+At a high level, the workflow will be as follows. First, the pricefeed module creates an IBC packet to request data from BandChain. Then, relayers will pick up the IBC packet and relay it on BandChain.
+
+After BandChain processes the request, it will send an acknowledgement message along with `request_id` back. And, when the result of the request is finalized, BandChain will send a new IBC packet that contains the final data back. Relayers will listen and pick up those packets and relay them to the smart contract.
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/introduction.md]: smart contract -> sth

Comment (colmazia) [reply docs/integration-tools/pricefeed/introduction.md]: please check if this paragraph is in other documentation such as bootcamp, project's git

File: docs/integration-tools/pricefeed/introduction.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+##### Symbol Requests 
+
+The `symbol-requests` command enables users to retrieve information about all symbol requests that are save in this Cosmos SDK application.
+
+```
+oracle-consumerd query pricefeed symbol-requests
+```
+
+##### Price
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/introduction.md]: Optional: bandchain docs website can't handle 5 # properly

File: docs/integration-tools/pricefeed/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+...
+genesis:
+  app_state:
+    gov:
+      voting_params:
+        voting_period: "40s"
+```
+
+
+### Step 1: Import pricefeed moudle to your cosmos app
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/getting-started.md]: module

File: docs/integration-tools/pricefeed/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+As tendermint is no longer being developed, the pricefeed module now uses the version implemented by informalsystems. Therefore, to replace the tendermint version, kindly add this line in `example/go.mod`.
+
+```
+replace (
+    ...
+    github.com/tendermint/tendermint => github.com/informalsystems/tendermint v0.34.26
+)
+```
+
+##### Replace to use ibc-go v5
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/getting-started.md]: 5 # not look great on bandchain docs website

File: docs/integration-tools/pricefeed/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+```
+replace (
+    ...
+    github.com/tendermint/tendermint => github.com/cometbft/cometbft v0.34.27
+)
+```
+
+#### Replace to use ibc-go v5
+
+To ensure compatibility with the pricefeed module, kindly update the ibc-go version to v5 by replacing
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: Can't v6 connect with v5?

File: docs/integration-tools/pricefeed/example.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
+<!--
+order: 3
+-->
+
+# Example Use Cases
+
+By utilizing pricefeed module in your cosmos-sdk application, you have the ability to request price data from BandChain via IBC. This creates a chance to obtain a variety of real-world data, including cryptocurrency and stock prices. Please take a look at the following example of an oracle-consumer chain that requests data from BandChain for use in their module.
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/example.md]: to use*

File: docs/integration-tools/pricefeed/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+genesis:
+  app_state:
+    gov:
+      voting_params:
+        voting_period: "40s"
+```
+
+## Step 2: Initiate source channel and symbol requests by Ignite
+
+To utilize the Ignite feature to replace the genesis state, insert the code shown below into the `config.yml` file. and restat the chin by using `ignite chain serve -r -v` command.
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/getting-started.md]: restart the chain*

File: docs/integration-tools/pricefeed/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+To ensure compatibility with the pricefeed module, kindly update the ibc-go version to v5 by replacing
+
+```go
+require (
+    ...
+    github.com/cosmos/ibc-go/v6 v6.1.0
+)
+```
+
+To
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/getting-started.md]: With*

File: docs/integration-tools/pricefeed/introduction.md [hunk 11]
```diff
@@ -2,4 +2,221 @@
...
+  option (gogoproto.goproto_stringer) = false;
+
+  uint64 ask_count = 1;
+  uint64 min_count = 2;
+  uint64 min_ds_count = 3;
+  uint64 prepare_gas_base = 4;
+  uint64 prepare_gas_each = 5;
+  uint64 execute_gas_base = 6;
+  uint64 execute_gas_each = 7;
+  string source_channel = 8;
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/introduction.md:52]: Just a quick notice. So every request has to use the same value of these params. Is it better for it to be flexibly set?

File: docs/integration-tools/pricefeed/introduction.md [hunk 12]
```diff
@@ -2,4 +2,221 @@
...
+}
+```
+
+The example of submit and vote the proposal is demonstrated in the CLI section.
+
+## CLI
+
+A user can query and interact with the pricefeed module using the CLI.
+
+> Note: This example use `oracle-consumerd` as a command-line interface (CLI) from [oracle consumer chain](https://). Please replace it with your own cosmos app.
```
Comment (colmazia) [inline docs/integration-tools/pricefeed/introduction.md]: there is no url

File: docs/integration-tools/pricefeed/introduction.md (hunk 11 above)
Comment (satawatnack) [reply docs/integration-tools/pricefeed/introduction.md:52]: I think it's ok for this version

File: docs/integration-tools/pricefeed/introduction.md (hunk 12 above)
Comment (colmazia) [reply docs/integration-tools/pricefeed/introduction.md]: So there is no need for the url? @satawatnack

Comment (satawatnack) [reply docs/integration-tools/pricefeed/introduction.md]: Screenshot 2566-04-21 at 16 15 58
already add this

**Review by colmazia — APPROVED** [review]

File: docs/integration-tools/pricefeed/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,355 @@
...
+#### Submit proposal
+
+```
+oracle-consumerd tx gov submit-legacy-proposal param-change param-change-proposal.json --from alice
+```
+
+#### Vote the proposal
+
+```
+exampled tx gov vote 1 yes --from alice
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md:293]: exampled?

File: docs/integration-tools/pricefeed/getting-started.md [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
+    }
+  ],
+  "deposit": "10000000stake"
+}
+```
+
+#### Submit proposal
+
+```
+oracle-consumerd tx gov submit-legacy-proposal param-change param-change-proposal.json --from alice
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: should be exampled?

File: docs/integration-tools/pricefeed/getting-started.md [hunk 15]
```diff
@@ -0,0 +1,355 @@
...
+```
+
+```
+exampled tx gov vote 1 yes --from bob
+```
+
+
+### Step 5.2: Open update symbol request proposal and vote
+
+The purpose of this proposal is to request price data from BandChain at `block_interval` specified in the proposal. If the proposal is approved, the pricefeed module will retrieve the data and store the response on the consumer chain.
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md:320]: can't find a note about block_interval being zero. please help check.

**Comment by RogerKSI** [comment]
please help reverify with our latest code.

File: docs/integration-tools/pricefeed/example.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+# Example Use Cases
+
+By utilizing pricefeed module in your cosmos-sdk application, you have the ability to request price data from BandChain via IBC. This creates a chance to obtain a variety of real-world data, including cryptocurrency and stock prices. Please take a look at the following example of an oracle-consumer chain that requests data from BandChain to use in their module.
+
+
+---
+
+### oracle-consumer
+
+The oracle-consumer is an application of the Cosmos SDK that demonstrates the use of the [pricefeed-module](https://github.com/bandprotocol/oracle-consumer/tree/main/docs/x/pricefeed.md) implemented by BandProtocol. This module allows other Cosmos SDK applications to easily obtain data from BandChain through IBC.
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/example.md]: link doesn't work. please check.

File: docs/integration-tools/pricefeed/getting-started.md [hunk 17]
```diff
@@ -0,0 +1,382 @@
...
+Additionally, modify the ibc-go dependency in both the go.mod and app.go files, replacing the version `v6.1.0` from the repository github.com/cosmos/ibc-go/v6 with version `v5.2.0` from the repository github.com/cosmos/ibc-go/v5.
+
+```go
+require (
+    ...
+    github.com/cosmos/ibc-go/v5 v5.2.0
+)
+```
+
+### Replace to use tendermint that develop by informalsystems
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md:95]: I think we should just say like 'Replace tendermint with cometbft'

File: docs/integration-tools/pricefeed/getting-started.md [hunk 18]
```diff
@@ -0,0 +1,382 @@
...
+)
+```
+
+Then run `go mod tidy` to update all module packages.
+
+
+<!-- ### Install pricefeed package
+
+```
+go install github.com/bandprotocol/oracle-consumer
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: remove if it isn't used anymore.

File: docs/integration-tools/pricefeed/getting-started.md [hunk 19]
```diff
@@ -0,0 +1,382 @@
...
+    pricefeed "github.com/bandprotocol/oracle-consumer/x/pricefeed"
+)
+
+ModuleBasics = module.NewBasicManager(
+    ...
+    pricefeed.AppModuleBasic{},
+)
+```
+
+#### Add pricefeed keeper type in `BandApp`
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: should not be bandapp?

File: docs/integration-tools/pricefeed/getting-started.md [hunk 20]
```diff
@@ -0,0 +1,382 @@
...
+
+```go
+import (
+    ...
+    pricefeedkeeper "github.com/bandprotocol/oracle-consumer/x/pricefeed/keeper"
+)
+
+type BandApp struct {
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: ditto.

File: docs/integration-tools/pricefeed/getting-started.md [hunk 21]
```diff
@@ -0,0 +1,382 @@
...
+```
+
+Once you have completed the addition of the pricefeed module in the app.go file, execute the command `go mod tidy` to import and update the necessary modules.
+
+Now have completed importing the pricefeed module and can now execute the chain by running this command :tada:
+```
+ignite chain serve -v
+```
+
+## Step 4: Setup a relayer
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: TBD: Those two docs are intended to use for IBC tx between wasm and BandChain.
- Should we update it to be more common?
- or, Should we link it official document?
- ...

File: docs/integration-tools/pricefeed/getting-started.md [hunk 22]
```diff
@@ -0,0 +1,382 @@
...
+genesis:
+  app_state:
+    ...
+    pricefeed:
+        params:
+            source_channel: "channel-0"
+        symbol_requests: [{"symbol": "BAND", "oracle_script_id": 396, "block_interval":  40}]
+```
+
+## Step 3: Import pricefeed module to your cosmos app
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: No step 2?

File: docs/integration-tools/pricefeed/getting-started.md (hunk 19 above)
Comment (satawatnack) [reply docs/integration-tools/pricefeed/getting-started.md:143]: Screenshot 2566-05-15 at 16 52 34

why we don't use BandApp like the real code

File: docs/integration-tools/pricefeed/getting-started.md [hunk 23]
```diff
@@ -0,0 +1,375 @@
...
+```go
+require (
+    ...
+    github.com/cosmos/ibc-go/v5 v5.2.0
+)
+```
+
+### Replace tendermint with cometbft 
+
+As tendermint is no longer being developed, the pricefeed module now uses the version implemented by cometbft. Therefore, to replace the tendermint version, kindly add this line in `go.mod`.
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md:97]: pls remove the first sentence as well. as we aren't quite sure what happen

File: docs/integration-tools/pricefeed/getting-started.md [hunk 24]
```diff
@@ -0,0 +1,375 @@
...
+The second step is to set up a relayer to listen and relay IBC packets between a your chain and BandChain.
+
+Here are the simple guides for setting up a relayer.
+
+- [Hermes relayer](https://github.com/bandprotocol/cw-band/blob/main/docs/setup_relayer_hermes.md)
+- [Go relayer](https://github.com/bandprotocol/cw-band/blob/main/docs/setup_relayer_go-relayer.md)
+
+## Step 4 (optional): Open proposal for change params and update symbol requests
+
+Since you have already configured the symbol requests and source-channel in the `config.yml` file during the [step 2](#step-2-initiate-source-channel-and-symbol-requests-by-ignite) , you may skip this particular step.
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/getting-started.md]: hyperlink doesn't work

File: docs/integration-tools/pricefeed/introduction.md [hunk 25]
```diff
@@ -2,4 +2,224 @@
...
+    ```
+    oracle-consumerd tx gov submit-legacy-proposal update-symbol-request update_symbol_requests.json --from alice
+    ```
+
+### Another way to initiate source channel and symbol requests
+
+To utilize the Ignite feature to replace the genesis state, insert the code shown below into the `config.yml` file. and restart the chain by using `ignite chain serve -r -v` command.
+
+```yml
+pricefeed:
```
Comment (RogerKSI) [inline docs/integration-tools/pricefeed/introduction.md]: no need to have it on top?
```
genesis:
  app_state:
    pricefeed:
```

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**